xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm

```diff
@@ -120,20 +120,22 @@
     ##   #if($!tags.size() == 0)
     ##     No tags
     ##   #else
-    $services.localization.render('core.tags.list.label')
-    #foreach($tag in $tags)
-        #displayTag($tag)
-    #end
     ##   #end
-    #if($hasedit)
-        #if($xwiki.tag)
-          <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-        #else
-            ## TODO
+    <div class="doc-tags #if(!$hasedit)hidden#end" id="${tagsId}">
+        $services.localization.render('core.tags.list.label')
+        #if($hasedit || !$tags.isEmpty())
+            #foreach($tag in $tags)
+                #displayTag($tag)
+            #end
+            #if($hasedit && $xwiki.tag)
+                <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
+            #else
+                ## TODO
```

Comment:
I know you haven't added it, but IMO this else part should be removed. It's useless and the TODO comment doesn't provide any info about what should be done next.

Reply:
check again